iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -699,6 +663,55 @@ private async Task EnsureOpenedAsync(bool withRetry, CancellationToken cancellat
             }
         }
 
+        public override async Task CloseAsync(CancellationToken cancellationToken)
+        {
+            await _handlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+            try
+            {
+                if (!_openCalled)
+                {
+                    return;
+                }
+
+                if (Logging.IsEnabled)
+                    Logging.Enter(this, cancellationToken, nameof(CloseAsync));
+
+                _handleDisconnectCts.Cancel();
+                await base.CloseAsync(cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, cancellationToken, nameof(CloseAsync));
+
+                _handlerSemaphore?.Release();
+                Dispose(true);
+            }
+        }
+
+        private async Task EnsureOpenedAsync()
+        {
+            try
+            {
+                await _handlerSemaphore.WaitAsync().ConfigureAwait(false);
+                if (!_opened)
+                {
+                    if (!_openCalled)
+                    {
+                        throw new InvalidOperationException($"Explicit call to {nameof(OpenAsync)} is missing before this operation.");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Transport disconnected, call to {nameof(OpenAsync)} is required before this operation.");
```

Comment:
```suggestion
                        throw new InvalidOperationException($"The transport has disconnected; call {'nameof(OpenAsync)}' to reconnect.");
```

Reply:
FWIW, I don't think we support a call to OpenAsync again (yet). It is in our list of things to do, but I'd be surprised if it works already.

Reply:
fixed in latest commit 
https://github.com/Azure/azure-iot-sdk-csharp/pull/2632/commits/47c9ce49828b188a309c818f8486485e2a38adbe